src/runtime/arch_tinygoriscv32.go

```diff
@@ -2,7 +2,7 @@
 
 package runtime
 
-const GOARCH = "arm" // riscv pretends to be arm
+const GOARCH = "riscv"
```

Comment:
Was this change intentional? I think the `GOARCH` value here should be kept in sync with the `GOARCH` in the target JSON file (targets/riscv.json) and with the build tag used there. Because those other two cannot be changed yet because of backwards compatibility, I think this should be kept at `"arm"`, even if that's incorrect. We can change that once we drop Go support before 1.14.

Reply:
Yes, for architecture without floating-point extension (like the hifive1b) it is fine to use the arm architecture, however with the floating-point extension (the maixbit), LLVM produces errors (see https://github.com/tinygo-org/tinygo/pull/1220#issuecomment-655605906) when using the LLVM intrinsic functions in `runtime/math.go`. These functions are used only on targets with the arm or wasm GOARCH but do not seem supported on RISC-V with floating point extension. So changing the GOARCH value is only necessary to avoid using the LLVM intrinsics in `runtime/math.go`. We could check with another variable to avoid using those intrinsic functions but I thought it would be cleaner using the GOARCH and since the CI seemed to pass even on the older Go versions I thought it didn't impact backwards compatibility. How do you suggest to avoid using the LLVM intrinsics without changing the GOARCH?
https://github.com/tinygo-org/tinygo/blob/a9ba6ebad9ad85cacd9674a1af9229489a811f68/src/runtime/math.go#L58-L64

Reply:
Okay, that's a good reason to change the `GOARCH` value.

Ideally we should be checking which floating point intrinsics are supported on RISC-V so that they can be used directly. Using those intrinsics may be significantly faster than calling math functions.

That said, I do believe this is a RISC-V backend bug. It should create calls to libc functions (such as `ceil`) instead of crashing.